File: _includes/code/java-v6/src/test/java/TokenizationTest.java
// START AccentFoldingCreateCollection
// START CustomStopwordsCreate
import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.Property;
import io.weaviate.client6.v1.api.collections.TextAnalyzer;
import io.weaviate.client6.v1.api.collections.Tokenization;
import io.weaviate.client6.v1.api.collections.VectorConfig;
// END AccentFoldingCreateCollection
// END CustomStopwordsCreate

// START TokenizeEndpointFreeform
import io.weaviate.client6.v1.api.tokenize.TokenizeResponse;
// END TokenizeEndpointFreeform

import io.weaviate.client6.v1.api.collections.CollectionHandle;
import io.weaviate.client6.v1.api.collections.query.Filter;
import io.weaviate.client6.v1.api.collections.query.Metadata;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.List;
import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Combined Java v6 test suite covering all v1.37 tokenization features:
 * accent folding, custom stopword presets, and the /v1/tokenize endpoint.
 */
class TokenizationTest {

  private static WeaviateClient client;

  @BeforeAll
  public static void beforeAll() throws IOException {
    client = WeaviateClient.connectToLocal();
  }

  @AfterAll
  public static void afterAll() throws Exception {
    if (client != null) {
      client.close();
    }
  }

  // ============================================================
  // Accent folding (asciiFold / asciiFoldIgnore)
  // ============================================================
  @Test
  void testAccentFolding() throws IOException {
    final String COLLECTION = "AccentFoldingDemo";
    client.collections.delete(COLLECTION);
    try {
      // START AccentFoldingCreateCollection
      client.collections.create(COLLECTION, c -> c
          .properties(
              Property.text("text_default",
                  p -> p.tokenization(Tokenization.WORD)),
              Property.text("text_folded",
                  p -> p.tokenization(Tokenization.WORD)
                      .textAnalyzer(TextAnalyzer.of(t -> t.foldAscii(true)))),
              Property.text("text_folded_keep_e",
                  p -> p.tokenization(Tokenization.WORD)
                      .textAnalyzer(TextAnalyzer.of(t -> t
                          .foldAscii(true)
                          .keepAscii("é")))))
          .vectorConfig(VectorConfig.selfProvided()));
      // END AccentFoldingCreateCollection

      // START AccentFoldingAddObjects
      CollectionHandle<Map<String, Object>> products =
          client.collections.use(COLLECTION);

      List<String> testStrings = List.of(
          "Café Crème Bio",
          "Łódź Ceramics",
          "São Paulo Sandals",
          "Müller Bräu");

      for (String text : testStrings) {
        products.data.insert(Map.of(
            "text_default", text,
            "text_folded", text,
            "text_folded_keep_e", text));
      }
      // END AccentFoldingAddObjects

      // START AccentFoldingFilter
      String[] queries = {"cafe", "Café", "lodz", "sao paulo", "muller"};
      String[] properties = {"text_default", "text_folded", "text_folded_keep_e"};

      for (String query : queries) {
        System.out.println("\nQuery: \"" + query + "\"");
        for (String prop : properties) {
          var response = products.query.fetchObjects(
              q -> q.filters(Filter.property(prop).eq(query)));
          var matches = response.objects().stream()
              .map(o -> (String) o.properties().get(prop))
              .toList();
          System.out.println(
              "  " + prop + ": " + (matches.isEmpty() ? "no match" : matches));
        }
      }
      // END AccentFoldingFilter

      // Test: "cafe" matches folded but not default or keep_e
      var r = products.query.fetchObjects(
          q -> q.filters(Filter.property("text_folded").eq("cafe")));
      assertThat(r.objects()).hasSize(1);
      assertThat((String) r.objects().get(0).properties().get("text_folded"))
          .isEqualTo("Café Crème Bio");

      r = products.query.fetchObjects(
          q -> q.filters(Filter.property("text_default").eq("cafe")));
      assertThat(r.objects()).isEmpty();

      r = products.query.fetchObjects(
          q -> q.filters(Filter.property("text_folded_keep_e").eq("cafe")));
      assertThat(r.objects()).isEmpty();

      // Test: "Café" matches all three
      for (String prop : properties) {
        var rr = products.query.fetchObjects(
            q -> q.filters(Filter.property(prop).eq("Café")));
        assertThat(rr.objects())
            .as("Expected 'Café' to match on %s", prop)
            .hasSize(1);
      }

      // Test: "lodz" matches folded and keep_e but not default
      r = products.query.fetchObjects(
          q -> q.filters(Filter.property("text_folded").eq("lodz")));
      assertThat(r.objects()).hasSize(1);
      assertThat((String) r.objects().get(0).properties().get("text_folded"))
          .contains("Łódź");

      r = products.query.fetchObjects(
          q -> q.filters(Filter.property("text_default").eq("lodz")));
      assertThat(r.objects()).isEmpty();

      r = products.query.fetchObjects(
          q -> q.filters(Filter.property("text_folded_keep_e").eq("lodz")));
      assertThat(r.objects()).hasSize(1);

      // Test: "muller" matches folded but not default
      r = products.query.fetchObjects(
          q -> q.filters(Filter.property("text_folded").eq("muller")));
      assertThat(r.objects()).hasSize(1);
      assertThat((String) r.objects().get(0).properties().get("text_folded"))
          .contains("Müller");

      r = products.query.fetchObjects(
          q -> q.filters(Filter.property("text_default").eq("muller")));
      assertThat(r.objects()).isEmpty();
    } finally {
      client.collections.delete(COLLECTION);
    }
  }

  // ============================================================
  // Custom and per-property stopword presets
  // ============================================================
  @Test
  void testCustomStopwords() throws IOException {
    final String COLLECTION = "StopwordsDemo";
    client.collections.delete(COLLECTION);
    try {
      // START CustomStopwordsCreate
      Map<String, List<String>> presets = Map.of(
          "fr", List.of("le", "la", "les", "un", "une", "des", "du", "de", "et"));

      client.collections.create(COLLECTION, c -> c
          .invertedIndex(idx -> idx.stopwordPresets(presets))
          .properties(
              Property.text("name_en",
                  p -> p.tokenization(Tokenization.WORD)
                      .textAnalyzer(TextAnalyzer.of(t -> t.stopwordPreset("en")))),
              Property.text("name_fr",
                  p -> p.tokenization(Tokenization.WORD)
                      .textAnalyzer(TextAnalyzer.of(t -> t.stopwordPreset("fr")))))
          .vectorConfig(VectorConfig.selfProvided()));
      // END CustomStopwordsCreate

      // START CustomStopwordsAddObjects
      CollectionHandle<Map<String, Object>> products =
          client.collections.use(COLLECTION);

      products.data.insert(Map.of(
          "name_en", "The Blue Cup and the Bowl",
          "name_fr", "La Tasse Bleue et le Bol"));
      products.data.insert(Map.of(
          "name_en", "A Red Plate with the Saucer",
          "name_fr", "Une Assiette Rouge avec la Soucoupe"));
      // END CustomStopwordsAddObjects

      // START CustomStopwordsSearch
      var responseFr = products.query.bm25(
          "la tasse bleue et le bol",
          q -> q.queryProperties("name_fr").returnMetadata(Metadata.SCORE));

      System.out.println("French property search:");
      for (var o : responseFr.objects()) {
        System.out.println(
            "  " + o.properties().get("name_fr")
                + " (score: " + o.queryMetadata().score() + ")");
      }

      var responseEn = products.query.bm25(
          "la tasse bleue et le bol",
          q -> q.queryProperties("name_en").returnMetadata(Metadata.SCORE));

      System.out.println("\nEnglish property search:");
      for (var o : responseEn.objects()) {
        System.out.println(
            "  " + o.properties().get("name_en")
                + " (score: " + o.queryMetadata().score() + ")");
      }
      // END CustomStopwordsSearch

      // Test: French search finds the matching document
      assertThat(responseFr.objects()).hasSize(1);
      assertThat((String) responseFr.objects().get(0).properties().get("name_fr"))
          .isEqualTo("La Tasse Bleue et le Bol");
      assertThat(responseFr.objects().get(0).queryMetadata().score())
          .isPositive();

      // Test: English search returns no results (French words aren't in English data)
      var enOnEn = products.query.bm25(
          "la tasse bleue et le bol",
          q -> q.queryProperties("name_en"));
      assertThat(enOnEn.objects()).isEmpty();

      // Test: Searching French property for English content returns no results
      var enOnFr = products.query.bm25(
          "blue cup bowl",
          q -> q.queryProperties("name_fr"));
      assertThat(enOnFr.objects()).isEmpty();

      // Test: "la" alone shouldn't score on French property — it's a stopword
      var laResponse = products.query.bm25(
          "la",
          q -> q.queryProperties("name_fr"));
      assertThat(laResponse.objects()).isEmpty();
    } finally {
      client.collections.delete(COLLECTION);
    }
  }

  // ============================================================
  // /v1/tokenize endpoint — freeform
  // ============================================================
  @Test
  void testTokenizeFreeform() throws IOException {
    // START TokenizeEndpointFreeform
    // Ad-hoc tokenization with custom config
    TokenizeResponse result = client.tokenize.text(
        "The organic café crème blend",
        t -> t
            .tokenization(Tokenization.WORD)
            .textAnalyzer(TextAnalyzer.of(a -> a
                .foldAscii(true)
                .stopwordPreset("en"))));

    System.out.println("indexed: " + result.indexed());
    System.out.println("query:   " + result.query());
    // END TokenizeEndpointFreeform

    // Test: accent folding converts café→cafe, crème→creme
    assertThat(result.indexed()).contains("cafe");
    assertThat(result.indexed()).contains("creme");
    // Test: stopword "the" is in indexed but not in query
    assertThat(result.indexed()).contains("the");
    assertThat(result.query()).doesNotContain("the");
    // Test: non-stopwords are in both
    assertThat(result.indexed()).contains("organic");
    assertThat(result.query()).contains("organic");
  }

  // ============================================================
  // /v1/tokenize endpoint — with custom preset
  // ============================================================
  @Test
  void testTokenizeCustomPreset() throws IOException {
    // START TokenizeEndpointCustomPreset
    // Define a named "fr" preset and reference it from analyzer config.
    // stopwordPresets is mutually exclusive with stopwords — pass at most one.
    Map<String, List<String>> presets = Map.of(
        "fr", List.of("le", "la", "les", "un", "une", "des", "du", "de", "et"));

    TokenizeResponse result = client.tokenize.text(
        "La Tasse Bleue et le Bol",
        t -> t
            .tokenization(Tokenization.WORD)
            .textAnalyzer(TextAnalyzer.of(a -> a.stopwordPreset("fr")))
            .stopwordPresets(presets));

    System.out.println("indexed: " + result.indexed());
    System.out.println("query:   " + result.query());
    // END TokenizeEndpointCustomPreset

    // Test: French stopwords are indexed but removed from query
    assertThat(result.indexed()).contains("la", "et", "le");
    assertThat(result.query()).doesNotContain("la", "et", "le");
    assertThat(result.query()).contains("tasse", "bleue", "bol");
  }

  // ============================================================
  // /v1/tokenize endpoint — for an existing property's config
  // ============================================================
  @Test
  void testTokenizeForProperty() throws IOException {
    final String COLLECTION = "TokenizeDemo";
    client.collections.delete(COLLECTION);
    try {
      Map<String, List<String>> presets = Map.of(
          "fr", List.of("le", "la", "les", "un", "une", "des", "du", "de", "et"));

      client.collections.create(COLLECTION, c -> c
          .invertedIndex(idx -> idx.stopwordPresets(presets))
          .properties(
              Property.text("name_fr",
                  p -> p.tokenization(Tokenization.WORD)
                      .textAnalyzer(TextAnalyzer.of(t -> t.stopwordPreset("fr")))))
          .vectorConfig(VectorConfig.selfProvided()));

      // START TokenizeEndpointProperty
      // Tokenize using an existing property's configuration
      TokenizeResponse result = client.tokenize.forProperty(
          "La Tasse Bleue et le Bol", COLLECTION, "name_fr");

      System.out.println("indexed: " + result.indexed());
      System.out.println("query:   " + result.query());
      // END TokenizeEndpointProperty

      // Test: all words are indexed (stopwords are still stored)
      assertThat(result.indexed()).contains("la", "et", "le");
      assertThat(result.indexed()).contains("tasse", "bleue", "bol");
      // Test: French stopwords are removed from query tokens
      assertThat(result.query()).doesNotContain("la", "et", "le");
      // Test: non-stopwords remain in query tokens
      assertThat(result.query()).contains("tasse", "bleue", "bol");
      assertThat(result.indexed()).hasSize(6);
      assertThat(result.query()).hasSize(3);
    } finally {
      client.collections.delete(COLLECTION);
    }
  }
}

File: _includes/code/csharp/TokenizationTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Weaviate.Client;
using Weaviate.Client.Models;
using Xunit;

namespace WeaviateProject.Tests;

/// <summary>
/// Combined C# test suite covering all v1.37 tokenization features:
/// accent folding, custom stopword presets, and the /v1/tokenize endpoint.
/// </summary>
public class TokenizationTest : IAsyncLifetime
{
    private WeaviateClient client;

    public async Task InitializeAsync()
    {
        client = await Connect.Local();
    }

    public Task DisposeAsync() => Task.CompletedTask;

    // ============================================================
    // Accent folding (asciiFold / asciiFoldIgnore)
    // ============================================================
    [Fact]
    public async Task TestAccentFolding()
    {
        const string COLLECTION = "AccentFoldingDemo";
        if (await client.Collections.Exists(COLLECTION))
            await client.Collections.Delete(COLLECTION);

        try
        {
            // START AccentFoldingCreateCollection
            await client.Collections.Create(
                new CollectionCreateParams
                {
                    Name = "AccentFoldingDemo",
                    Properties =
                    [
                        new Property
                        {
                            Name = "text_default",
                            DataType = DataType.Text,
                            PropertyTokenization = PropertyTokenization.Word,
                        },
                        new Property
                        {
                            Name = "text_folded",
                            DataType = DataType.Text,
                            PropertyTokenization = PropertyTokenization.Word,
                            TextAnalyzer = new TextAnalyzerConfig
                            {
                                AsciiFold = new AsciiFoldConfig(),
                            },
                        },
                        new Property
                        {
                            Name = "text_folded_keep_e",
                            DataType = DataType.Text,
                            PropertyTokenization = PropertyTokenization.Word,
                            TextAnalyzer = new TextAnalyzerConfig
                            {
                                AsciiFold = new AsciiFoldConfig(Ignore: ["é"]),
                            },
                        },
                    ],
                    VectorConfig = new VectorConfigList
                    {
                        Configure.Vector("default", v => v.SelfProvided()),
                    },
                }
            );
            // END AccentFoldingCreateCollection

            // START AccentFoldingAddObjects
            var products = client.Collections.Use("AccentFoldingDemo");

            string[] testStrings =
            [
                "Café Crème Bio",
                "Łódź Ceramics",
                "São Paulo Sandals",
                "Müller Bräu",
            ];

            foreach (var text in testStrings)
            {
                await products.Data.Insert(
                    new
                    {
                        text_default = text,
                        text_folded = text,
                        text_folded_keep_e = text,
                    }
                );
            }
            // END AccentFoldingAddObjects

            // START AccentFoldingFilter
            string[] queries = ["cafe", "Café", "lodz", "sao paulo", "muller"];
            string[] properties = ["text_default", "text_folded", "text_folded_keep_e"];

            foreach (var query in queries)
            {
                Console.WriteLine($"\nQuery: \"{query}\"");
                foreach (var prop in properties)
                {
                    var response = await products.Query.FetchObjects(
                        filters: Filter.Property(prop).IsEqual(query)
                    );
                    var matches = response
                        .Objects.Select(o => (string)o.Properties[prop])
                        .ToList();
                    Console.WriteLine(
                        $"  {prop}: {(matches.Count == 0 ? "no match" : string.Join(", ", matches))}"
                    );
                }
            }
            // END AccentFoldingFilter

            // Test: "cafe" matches folded but not default or keep_e
            var r = await products.Query.FetchObjects(
                filters: Filter.Property("text_folded").IsEqual("cafe")
            );
            Assert.Single(r.Objects);
            Assert.Equal("Café Crème Bio", (string)r.Objects.First().Properties["text_folded"]);

            r = await products.Query.FetchObjects(
                filters: Filter.Property("text_default").IsEqual("cafe")
            );
            Assert.Empty(r.Objects);

            r = await products.Query.FetchObjects(
                filters: Filter.Property("text_folded_keep_e").IsEqual("cafe")
            );
            Assert.Empty(r.Objects);

            // Test: "Café" matches all three
            foreach (var prop in properties)
            {
                var rr = await products.Query.FetchObjects(
                    filters: Filter.Property(prop).IsEqual("Café")
                );
                Assert.True(rr.Objects.Any(), $"Expected 'Café' to match on {prop}");
            }

            // Test: "lodz" matches folded and keep_e but not default
            r = await products.Query.FetchObjects(
                filters: Filter.Property("text_folded").IsEqual("lodz")
            );
            Assert.Single(r.Objects);
            Assert.Contains("Łódź", (string)r.Objects.First().Properties["text_folded"]);

            r = await products.Query.FetchObjects(
                filters: Filter.Property("text_default").IsEqual("lodz")
            );
            Assert.Empty(r.Objects);

            r = await products.Query.FetchObjects(
                filters: Filter.Property("text_folded_keep_e").IsEqual("lodz")
            );
            Assert.Single(r.Objects);

            // Test: "muller" matches folded but not default
            r = await products.Query.FetchObjects(
                filters: Filter.Property("text_folded").IsEqual("muller")
            );
            Assert.Single(r.Objects);
            Assert.Contains("Müller", (string)r.Objects.First().Properties["text_folded"]);

            r = await products.Query.FetchObjects(
                filters: Filter.Property("text_default").IsEqual("muller")
            );
            Assert.Empty(r.Objects);
        }
        finally
        {
            if (await client.Collections.Exists(COLLECTION))
                await client.Collections.Delete(COLLECTION);
        }
    }

    // ============================================================
    // Custom and per-property stopword presets
    // ============================================================
    [Fact]
    public async Task TestCustomStopwords()
    {
        const string COLLECTION = "StopwordsDemo";
        if (await client.Collections.Exists(COLLECTION))
            await client.Collections.Delete(COLLECTION);

        try
        {
            // START CustomStopwordsCreate
            var presets = new Dictionary<string, IList<string>>
            {
                ["fr"] = ["le", "la", "les", "un", "une", "des", "du", "de", "et"],
            };

            await client.Collections.Create(
                new CollectionCreateParams
                {
                    Name = "StopwordsDemo",
                    InvertedIndexConfig = new InvertedIndexConfig { StopwordPresets = presets },
                    Properties =
                    [
                        new Property
                        {
                            Name = "name_en",
                            DataType = DataType.Text,
                            PropertyTokenization = PropertyTokenization.Word,
                            TextAnalyzer = new TextAnalyzerConfig { StopwordPreset = "en" },
                        },
                        new Property
                        {
                            Name = "name_fr",
                            DataType = DataType.Text,
                            PropertyTokenization = PropertyTokenization.Word,
                            TextAnalyzer = new TextAnalyzerConfig { StopwordPreset = "fr" },
                        },
                    ],
                    VectorConfig = new VectorConfigList
                    {
                        Configure.Vector("default", v => v.SelfProvided()),
                    },
                }
            );
            // END CustomStopwordsCreate

            // START CustomStopwordsAddObjects
            var products = client.Collections.Use("StopwordsDemo");

            await products.Data.Insert(
                new
                {
                    name_en = "The Blue Cup and the Bowl",
                    name_fr = "La Tasse Bleue et le Bol",
                }
            );
            await products.Data.Insert(
                new
                {
                    name_en = "A Red Plate with the Saucer",
                    name_fr = "Une Assiette Rouge avec la Soucoupe",
                }
            );
            // END CustomStopwordsAddObjects

            // START CustomStopwordsSearch
            var responseFr = await products.Query.BM25(
                query: "la tasse bleue et le bol",
                searchFields: ["name_fr"],
                returnMetadata: MetadataOptions.Score
            );

            Console.WriteLine("French property search:");
            foreach (var o in responseFr.Objects)
            {
                Console.WriteLine(
                    $"  {o.Properties["name_fr"]} (score: {o.Metadata.Score})"
                );
            }

            var responseEn = await products.Query.BM25(
                query: "la tasse bleue et le bol",
                searchFields: ["name_en"],
                returnMetadata: MetadataOptions.Score
            );

            Console.WriteLine("\nEnglish property search:");
            foreach (var o in responseEn.Objects)
            {
                Console.WriteLine(
                    $"  {o.Properties["name_en"]} (score: {o.Metadata.Score})"
                );
            }
            // END CustomStopwordsSearch

            // Test: French search finds the matching document
            Assert.Single(responseFr.Objects);
            Assert.Equal(
                "La Tasse Bleue et le Bol",
                (string)responseFr.Objects.First().Properties["name_fr"]
            );
            Assert.True(responseFr.Objects.First().Metadata.Score > 0);

            // Test: English property doesn't match French content
            var enOnEn = await products.Query.BM25(
                query: "la tasse bleue et le bol",
                searchFields: ["name_en"]
            );
            Assert.Empty(enOnEn.Objects);

            // Test: French property doesn't match English content
            var enOnFr = await products.Query.BM25(
                query: "blue cup bowl",
                searchFields: ["name_fr"]
            );
            Assert.Empty(enOnFr.Objects);

            // Test: "la" alone shouldn't score on French property — it's a stopword
            var laResponse = await products.Query.BM25(
                query: "la",
                searchFields: ["name_fr"]
            );
            Assert.Empty(laResponse.Objects);
        }
        finally
        {
            if (await client.Collections.Exists(COLLECTION))
                await client.Collections.Delete(COLLECTION);
        }
    }

    // ============================================================
    // /v1/tokenize endpoint — freeform
    // ============================================================
    [Fact]
    public async Task TestTokenizeFreeform()
    {
        // START TokenizeEndpointFreeform
        // Ad-hoc tokenization with custom config
        var result = await client.Tokenize.Text(
            text: "The organic café crème blend",
            tokenization: PropertyTokenization.Word,
            analyzerConfig: new TextAnalyzerConfig
            {
                AsciiFold = new AsciiFoldConfig(),
                StopwordPreset = "en",
            }
        );

        Console.WriteLine($"indexed: [{string.Join(", ", result.Indexed)}]");
        Console.WriteLine($"query:   [{string.Join(", ", result.Query)}]");
        // END TokenizeEndpointFreeform

        // Test: accent folding converts café→cafe, crème→creme
        Assert.Contains("cafe", result.Indexed);
        Assert.Contains("creme", result.Indexed);
        // Test: stopword "the" is in indexed but not in query
        Assert.Contains("the", result.Indexed);
        Assert.DoesNotContain("the", result.Query);
        // Test: non-stopwords are in both
        Assert.Contains("organic", result.Indexed);
        Assert.Contains("organic", result.Query);
    }

    // ============================================================
    // /v1/tokenize endpoint — with custom preset
    // ============================================================
    [Fact]
    public async Task TestTokenizeCustomPreset()
    {
        // START TokenizeEndpointCustomPreset
        // Define a named "fr" preset and reference it from analyzer config.
        // stopwordPresets is mutually exclusive with stopwords — pass at most one.
        var presets = new Dictionary<string, IList<string>>
        {
            ["fr"] = ["le", "la", "les", "un", "une", "des", "du", "de", "et"],
        };

        var result = await client.Tokenize.Text(
            text: "La Tasse Bleue et le Bol",
            tokenization: PropertyTokenization.Word,
            analyzerConfig: new TextAnalyzerConfig { StopwordPreset = "fr" },
            stopwordPresets: presets
        );

        Console.WriteLine($"indexed: [{string.Join(", ", result.Indexed)}]");
        Console.WriteLine($"query:   [{string.Join(", ", result.Query)}]");
        // END TokenizeEndpointCustomPreset

        // Test: French stopwords are indexed but removed from query
        Assert.Contains("la", result.Indexed);
        Assert.Contains("et", result.Indexed);
        Assert.Contains("le", result.Indexed);
        Assert.DoesNotContain("la", result.Query);
        Assert.DoesNotContain("et", result.Query);
        Assert.DoesNotContain("le", result.Query);
        Assert.Contains("tasse", result.Query);
        Assert.Contains("bleue", result.Query);
        Assert.Contains("bol", result.Query);
    }

    // ============================================================
    // /v1/tokenize endpoint — for an existing property's config
    // ============================================================
    [Fact]
    public async Task TestTokenizeForProperty()
    {
        const string COLLECTION = "TokenizeDemo";
        if (await client.Collections.Exists(COLLECTION))
            await client.Collections.Delete(COLLECTION);

        try
        {
            var presets = new Dictionary<string, IList<string>>
            {
                ["fr"] = ["le", "la", "les", "un", "une", "des", "du", "de", "et"],
            };

            await client.Collections.Create(
                new CollectionCreateParams
                {
                    Name = COLLECTION,
                    InvertedIndexConfig = new InvertedIndexConfig { StopwordPresets = presets },
                    Properties =
                    [
                        new Property
                        {
                            Name = "name_fr",
                            DataType = DataType.Text,
                            PropertyTokenization = PropertyTokenization.Word,
                            TextAnalyzer = new TextAnalyzerConfig { StopwordPreset = "fr" },
                        },
                    ],
                    VectorConfig = new VectorConfigList
                    {
                        Configure.Vector("default", v => v.SelfProvided()),
                    },
                }
            );

            // START TokenizeEndpointProperty
            // Tokenize using an existing property's configuration
            var collection = client.Collections.Use(COLLECTION);
            var result = await collection.Tokenize.Property(
                propertyName: "name_fr",
                text: "La Tasse Bleue et le Bol"
            );

            Console.WriteLine($"indexed: [{string.Join(", ", result.Indexed)}]");
            Console.WriteLine($"query:   [{string.Join(", ", result.Query)}]");
            // END TokenizeEndpointProperty

            // Test: all words are indexed (stopwords are still stored)
            Assert.Contains("la", result.Indexed);
            Assert.Contains("et", result.Indexed);
            Assert.Contains("le", result.Indexed);
            Assert.Contains("tasse", result.Indexed);
            Assert.Contains("bleue", result.Indexed);
            Assert.Contains("bol", result.Indexed);
            // Test: French stopwords are removed from query tokens
            Assert.DoesNotContain("la", result.Query);
            Assert.DoesNotContain("et", result.Query);
            Assert.DoesNotContain("le", result.Query);
            // Test: non-stopwords remain in query tokens
            Assert.Contains("tasse", result.Query);
            Assert.Contains("bleue", result.Query);
            Assert.Contains("bol", result.Query);
            Assert.Equal(6, result.Indexed.Count);
            Assert.Equal(3, result.Query.Count);
        }
        finally
        {
            if (await client.Collections.Exists(COLLECTION))
                await client.Collections.Delete(COLLECTION);
        }
    }
}
